boring-sys/patches/boring-pq.patch

```diff
@@ -4025,21 +4456,29 @@ index cafae9d17..074ac5906 100644
 +OPENSSL_EXPORT void KYBER768_marshal_public_key(
 +    uint8_t out[KYBER768_PUBLIC_KEY_BYTES], const struct KYBER768_public_key *in_pub);
 +
++// IPDWING_marshal_public_key serialises |in_pub| to |out|.
++OPENSSL_EXPORT void IPDWING_marshal_public_key(
++    uint8_t out[IPDWING_PUBLIC_KEY_BYTES], const struct IPDWING_public_key *in_pub);
++
 +// KYBER512_parse_public_key sets |*out| to the public-key encoded in |in|.
 +OPENSSL_EXPORT void KYBER512_parse_public_key(
 +    struct KYBER512_public_key *out, const uint8_t in[KYBER512_PUBLIC_KEY_BYTES]);
 +
 +// KYBER768_parse_public_key sets |*out| to the public-key encoded in |in|.
 +OPENSSL_EXPORT void KYBER768_parse_public_key(
 +    struct KYBER768_public_key *out, const uint8_t in[KYBER768_PUBLIC_KEY_BYTES]);
++
++// IPDWING_parse_public_key sets |*out| to the public-key encoded in |in|.
```

Comment:
Why do we have a parser function for the public key but not the ciphertext?

Reply:
Because the public key can hide precomputed values like the matrix A. (The patch doesn't do that though.)